### PR TITLE
Update charm to update the root kubeconfig with a current CA

### PR DESCRIPTION
**Pull request by addyess**
## Overview
Each config hook should do it's best to update the root kubeconfig to have a valid ca cert.

## Details
According to the bug [LP#2064305][], the charm code couldn't connect to the control-plane because 

```
2024-04-27 06:14:19 WARNING unit.kubernetes-control-plane/1.certificates-relation-changed logger.go:60 Unable to connect to the server: tls: failed to verify certificate: x509: certificate signed by unknown authority
```

This can be caused by the CA being invalid on the next charm hook.  This PR addresses the issue by ensuring the CA cert is correct on each possible reconcile loop

Draft until the following is merged
* https://github.com/charmed-kubernetes/charm-lib-kubernetes-snaps/pull/21

[LP#2064305]: https://bugs.launchpad.net/charm-kubernetes-master/+bug/2064305

### CODE REVIEW

File: requirements.txt
Comment (addyess) [inline requirements.txt:7]: Must revert this line once merged

File: src/charm.py
Comment (addyess) [inline src/charm.py:347]: the crux of the fix is to ensure the ca is updated in the kubeconfig

File: src/kubectl.py
Comment (addyess) [inline src/kubectl.py:75]: Extra debug on kubectl calls to print the stderr to the error log

File: src/charm.py
Comment (kwmonroe) [reply src/charm.py:347]: ok - so this works because we may not have the api server up yet, hence can't verify `kubectl get secrets`, and if the CA changes during this time, our "bootstrap" client config wasn't getting updated because we were checking for "if file dne".

is that scenario what you're fixing with this?

**Review by kwmonroe — APPROVED** [review]
i added a comment to the crux of this fix just to make sure i understand why we need this.  assuming i got that scenario right, this LGTM.

and thanks for the extra kubectl debug!

File: src/charm.py
Comment (addyess) [reply src/charm.py:347]: Right, according to the issue tracked in the bug, the certificate was invalid in the stack trace starting at 

```
auth_webhook.create_token(...)
   auth_webhook.get_token(...)
       auth_webhook.get_secrets(...)
            kubectl(...)
```

I surmised that the CA could be updated here to ensure we always have the right cert in place before checking the tokens.

**Pull request merged**
